data-prepper-plugins/s3-source/src/main/java/org/opensearch/dataprepper/plugins/source/SqsWorker.java

```diff
@@ -204,22 +214,37 @@ private List<DeleteMessageBatchRequestEntry> processS3EventNotificationRecords(f
 
         LOG.info("Received {} messages from SQS. Processing {} messages.", s3EventNotificationRecords.size(), parsedMessagesToRead.size());
 
+        List<DeleteMessageBatchRequestEntry> waitingForAcknowledgements = new ArrayList<>();
+        AcknowledgementSet acknowledgementSet = null;
+        if (endToEndAcknowledgementsEnabled) {
+            // Acknowledgement Set timeout is slightly smaller than the visibility timeout;
+            int timeout = (int) sqsOptions.getVisibilityTimeout().getSeconds() - 2;
+            acknowledgementSet = acknowledgementSetManager.create((result) -> {
+                acknowledgementSetCallbackCounter.increment();
```

Comment:
Is this a valuable metric for customers?

Reply:
May be not. It could be valuable to us. I can remove it if you think it's not going to be helpful.